ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiActiveContractsService.scala

```diff
@@ -42,6 +46,10 @@ private[apiserver] final class ApiActiveContractsService private (
           .validate(request.getFilter)
           .fold(Source.failed, backend.getActiveContracts(_, request.verbose))
           .via(logger.logErrorsOnStream)
+          .map(item => {
```

Comment:
Counting for all consumers might lead to metrics with ambiguous metrics when we have multiple consumers with varying behaviors in the application. Are you ok with this limitation or should we rather particularize the metric per consumer somehow (e.g. use the traceId)?

Reply:
You cannot expand the metrics namespace without serious consequences for the metrics gathering systems. Wavefront does not support well a dynamically growing metrics namespace (datadog also does not), we are also reaching the limit of what grafana is willing to accept. 
The solution would be to add labels, in dropwizard terms they are called tags. They will be supported in version 5.0 of dropwizard, which has been i n the making for the past 3 years.

Reply:
You're right that it's not ideal to have dynamic metrics namespace, but we already do have them in a lot of places. If this solution serves the need now, let's leave it like this and keep metric specialization in mind for future iterations.